week5/WordGraph.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Created by julia on 7/6/17.
+//
+
+#include "WordGraph.h"
+
+
+WordGraph::WordGraph(std::vector<std::string> &GivenDictionary) {
+    std::vector<int> connection (0);
+    for (int i = 0; i < 128; ++i) {
+        IsUsed_.push_back(false);
+        Connections_.push_back(connection);
+    }
+    int Size = GivenDictionary.size();
+    std::string first_word; // variable
+    std::string second_word; // variable
+    for (int i = 0; i < Size - 1; ++i) {
+        int index = 0;
+        first_word = GivenDictionary[i];
+        second_word = GivenDictionary[i + 1];
+        int border = std::min(first_word.length(), second_word.length());
+        while (index < border && first_word[index] == second_word[index]) { // before adding a connection
+            IsUsed_[(int)first_word[index]] = true;
+            ++index;
+        }
+        if (index < border) { // adding a connection
+            Connections_[(int)second_word[index]].push_back((int)first_word[index]);
+        }
+        while (index < first_word.length()) { // after adding a connection
+            IsUsed_[(int)first_word[index]] = true;
+            ++index;
+        }
+    }
+    if (Size != 0) {
+        first_word = GivenDictionary[Size - 1];
+        for (int i = 0; i < first_word.length(); ++i) {
+            IsUsed_[(int)first_word[i]] = true;
+        }
+    }
+}
+
+
+void WordGraph::DFS(std::vector<char>& answer, int starter, std::vector<bool>& trouble) {
```

Comment:
You can use the return for the success / error condition. It's a convention used sometimes. Otherwise you can just pass in a `bool *` / `bool &`